contracts/libraries/HyperdriveExecution.sol

```diff
@@ -0,0 +1,578 @@
+
+        // NOTE: Round down to underestimate the total governance fee.
+        //
+        // We calculate the flat portion of the governance fee as:
+        //
+        // governance_flat_fee = flat_fee * phi_gov
+        //                     = shares * phi_gov
+        //
+        // The totalGovernanceFee is the sum of the curve and flat governance fees.
+        totalGovernanceFee =
+            governanceCurveFee +
+            flatFee.mulDown(_governanceLPFee);
+    }
+
+    /// @dev Calculates the fees that go to the LPs and governance.
+    /// @dev See `lib/hyperdrive/contracts/src/internal/HyperdriveBase.sol`
+    ///      for more information.
+    /// @param _shareAmount The amount of shares exchanged for bonds.
+    /// @param _spotPrice The price without slippage of bonds in terms of base
+    ///         (base/bonds).
+    /// @param _vaultSharePrice The current vault share price (base/shares).
+    /// @return curveFee The curve fee. The fee is in terms of bonds.
+    /// @return governanceCurveFee The curve fee that goes to governance. The
+    ///         fee is in terms of bonds.
+    function _calculateFeesGivenShares(
+        uint256 _shareAmount,
+        uint256 _spotPrice,
+        uint256 _vaultSharePrice,
+        uint256 _curveFee,
+        uint256 _governanceLPFee
+    ) internal pure returns (uint256 curveFee, uint256 governanceCurveFee) {
+        // NOTE: Round up to overestimate the curve fee.
+        //
+        // Fixed Rate (r) = (value at maturity - purchase price)/(purchase price)
+        //                = (1-p)/p
+        //                = ((1 / p) - 1)
+        //                = the ROI at maturity of a bond purchased at price p
+        //
+        // Another way to think about it:
+        //
+        // p (spot price) tells us how many base a bond is worth -> p = base/bonds
+        // 1/p tells us how many bonds a base is worth -> 1/p = bonds/base
+        // 1/p - 1 tells us how many additional bonds we get for each
+        // base -> (1/p - 1) = additional bonds/base
+        //
+        // The curve fee is taken from the additional bonds the user gets for
+        // each base:
+        //
+        // curve fee = ((1 / p) - 1) * phi_curve * c * dz
+        //           = r * phi_curve * base/shares * shares
+        //           = bonds/base * phi_curve * base
+        //           = bonds * phi_curve
+        curveFee = (uint256(1e18).divUp(_spotPrice) - 1e18)
+            .mulUp(_curveFee)
+            .mulUp(_vaultSharePrice)
+            .mulUp(_shareAmount);
+
+        // NOTE: Round down to underestimate the governance curve fee.
+        //
+        // We leave the governance fee in terms of bonds:
+        // governanceCurveFee = curve_fee * phi_gov
+        //                    = bonds * phi_gov
+        governanceCurveFee = curveFee.mulDown(_governanceLPFee);
+    }
+
+    /// @notice Obtains the vaultSharePrice from the hyperdrive instance.
```

Comment:
Since this function (and the others below are internal), they aren't part of a public interface, and it would make sense to change the `@notice`s to `@dev`s. In the [NatSpec guide](https://docs.soliditylang.org/en/latest/natspec-format.html), `@notice` is defined as containing "Explain to an end user what this does," and in the case of internal functions, the end user is the developer.

Reply:
The intent with notice/dev usage in the library was to mimic their meanings within a contract context. As you and natspec mention, these are for external users, who for a library are developers.

That being said, natspec also says dev comments are to "Explain to a developer any extra details" yet our usage of them mostly involves high level descriptions of internal functions, with the details handled by block commenting within the function.

Regardless, I will do as you have asked and migrate all top level comments to @dev tags

Reply:
I hear you. The way that I've treated this in the past is that `@notice` is for things that are part of the external interface, but I get what you are saying.